extensions/spatial.sigmf-ext.md

```diff
@@ -0,0 +1,362 @@
+spherical systems have a specific relationship with the reported azimuth and
+elevation, which use the conventional geospatial definitions (degrees east of
+north, and degrees above horizon respectively).
+
+![SigMF Spatial Coordinate Reference System](spatial_crs.png)
+
+**Figure 1 - SigMF Spatial Extension Coordinate Reference Systems**
+
+As shown in Figure 1, the boresight of the aperture is defined as being in the
+direction of the positive X-axis. Based on this, as an example, a horizontal
+uniform linear array aperture would be defined along the Y-axis (see examples
+section below).
+
+## 0 Datatypes
+
+This extension defines the following datatypes:
+
+|type|long-form name|description|
+|----|--------------|-----------|
+|bearing|signal direction bearing|Quantitative representation of a direction with optional error.|
+|cartesian_point|cartesian position|Cartesian coordinate triplet [x, y, z]|
+
+### 0.1 The `bearing` Object
+
+The `bearing` field is used to describe relative one or two dimensional signal
+directions. The units are always in degrees; while it is legal for this field to
+have any value it is RECOMMENDED that it be between 0 and 360 or +/- 180.
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`azimuth`|false|float|degrees|Azimuth component of the direction in degrees increasing clockwise.|
+|`elevation`|false|float|degrees|Elevation component of the direction in degrees above horizon.|
+|`range`|false|float|meters|Line-of-sight slant range to emitter, if known, in meters.|
+|`az_error`|false|float|degrees|Error or uncertainty in the azimuth component.|
+|`el_error`|false|float|degrees|Error or uncertainty in the elevation component.|
+|`range_error`|false|float|meters|Error or uncertainty in the range component.|
+
+The `az_error`, `el_error`, and `range_error` fields SHOULD NOT be used if the
+corresponding field estimates are not included.
+
+```json
+  "bearing": {
+    "azimuth": 211.2,
+    "elevation": 15.0,
+    "range": 30,
+    "az_error": 0.1,
+    "el_error": 0.1,
+    "range_error": 0.1
+  }
+```
+
+### 0.2 The `cartesian_point` Object
+
+The `cartesian_point` field is used to describe a single point in the Cartesian
+coordinate reference system. This object is necessary to define the phase center
+geometries for multidimensional arrays, and other cartesian locations.
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`point`|true|float array|meters|Cartesian Point defined by a element array of [x,y,z] referenced to the `spatial` CRS.|
+
+
+## 1 Global
+
+The `spatial` extension adds the following fields to the `global` SigMF object:
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`num_elements`|true|int|N/A|Defines the number of phase centers / channels collected in the Collection or multichannel Dataset.|
+|`channel_num`|true|int|N/A|The channel number, shares index with `element_geometry`.|
```

Comment:
Now, *this* field would *not* apply to multi-channel `.sigmf-data` files, agreed? I suggest a JSON object in a collection be the one and only place that `num_elements` and `channel_num` be indicated.

Reply:
> Now, this field would not apply to multi-channel .sigmf-data files, agreed?

yep, this doesn't make much sense for multichannel datasets.

Some place needs to map each file to aperture arrays though, and in our work we have found that contextual information is of significant value to the individual files, so this is included here. 

multichannel datasets were sort of shoehorned in to this when they came into existence (they were never really a roadmapped feature, they just sort of came about). hrmm will need to think more. same issue with conditional requirement as-is.

Reply:
I have added clarification specifically for the case of multi-channel datasets. Even with multichannel datasets there is no guarantee that all of the data required for the spatial extension is contained in one file (e.g.: Two four channel radios that are synchronized and aligned with each other providing 4 phase-coherent channels in two separate 4 channel files), so this is only redundant in the specific (not recommended) case where all data is within one sigmf file.

Reply:
See above for the concept, the following excerpt holds true here as well I think.

> ultimately I think that the cost of a single redundant field is negligible especially considering it contains contextually useful information even in a standalone sense.